evm/spec/tables/memory.tex

```diff
@@ -43,8 +43,11 @@ \subsection{Memory}
   \item $e_i (e_i - 1) = 0$,
   \item $(1 - e_i) (a_{i + 1} - a_i) = 0$,
   \item $c_i < 2^{32}$.
+  \item $e_i (v_{i + 1}) < 2^{32}$
 \end{enumerate}
-The last constraint emulates a comparison between two addresses or timestamps by bounding their difference; this assumes that all addresses and timestamps fit in 32 bits and that the field is larger than that.
+The third constraint emulates a comparison between two addresses or timestamps by bounding their difference; this assumes that all addresses and timestamps fit in 32 bits and that the field is larger than that.
+
+The fourth constraint checks that the first value of a new address is indeed the smallest in history. Without this check, addresses could start at -1 (i.e. $p - 2$) and then increase properly.
```

Comment:
It sounds a bit like you are talking about the actual value, and the address itself here.
```suggestion
The fourth constraint checks that the virtual address of a new address is indeed the smallest in history. Without this check, addresses could start at -1 (i.e. $p - 2$) and then increase properly.
```